test/test_runner.c

```diff
@@ -88,7 +88,7 @@ static u32 AssignCostToRunner(void)
     u32 minCostProcess;
 
     if (gTestRunnerState.test->runner == &gAssumptionsRunner)
-        return TRUE;
+        return 0;
```

Comment:
Ah, I think I see what I messed up—this code looks like it was expecting to return `TRUE` if the test should run on this runner, and `FALSE` if not; but somewhere along the way it changed to returning the process to run on instead.

The reason why you'd want to run the assumptions on every runner is that the tests which are covered by the assumptions run on an arbitrary set of runners. So I wonder if this should say `return gTestRunnerI;` so that the test runs on all the runners (at zero cost, because we've skipped `estimateCost` on line 98, which is also important for keeping all the `processCosts` in sync).

I think atm if we have no tests except:
- an `ASSUMPTIONS` block which fails
- two tests that rely on that block

Then if we `make check -j2` the second test will run even though it shouldn't because `ASSUMPTIONS` and the first test will run on runner 0, and just the second test will run on runner 1 (`skipFilename` never gets set, because the `ASSUMPTIONS` doesn't run on that runner).

Reply:
By atm you mean with or without this PR's changes?

Reply:
With these changes. (It would be similar without, but swapping runners 0 and 1 because `return TRUE;` is really `return 1;`)

Reply:
I see. Now how to fix it : D

Reply:
You'd have to double-check (or I can do it later when I'm home from work), but I think just replacing `return 0;` with `return gTestRunnerI;` ought to do it :crossed_fingers:

Reply:
Looks like that worked (except with an extra `ASSUME failed`)

![image](https://github.com/rh-hideout/pokeemerald-expansion/assets/2904965/a719479d-7b50-4e63-af0f-65d9fa62b26c)

Reply:
Do we care about fixing this extra `ASSUME failed`?